sdk/eventhub/azure-eventhub/azure/eventhub/aio/_client_base_async.py

```diff
@@ -156,6 +229,13 @@ def __init__(
             fully_qualified_namespace=fully_qualified_namespace,
             eventhub_name=eventhub_name,
             credential=self._credential,
+            uamqp_transport=uamqp_transport,
```

Comment:
This needs to be passed in?

Reply:
ah yeah, this is then passed into Configuration inside the `ClientBase` constructor. And is used to build the hostname/custom_endpoint_address, which are different for uamqp vs. pyamqp

Reply:
I was figuring we'd just get the distinction from the transport that's passed in on line 233. Though it's internal so it doesn't matter too much.

Reply:
ah, originally, I wasn't going to pass in the uamqp_transport kwarg at all and do what you said. When I tried it, it seemed a little messier inside the ClientBase constructor, so I decided to just pass it in. Before:
``python
class ClientBase:
def __init__(...):
  # check for async AmqpTransport first
  self._amqp_transport = kwargs.pop("amqp_transport", None)
  uamqp_transport = False
  
  # if PyamqpTransportAsync
  if self._amqp_transport and self._amqp_transport.TIMEOUT_FACTOR == PyamqpTransport.TIMEOUT_FACTOR:
    uamqp_transport = True
  elif not self._amqp_transport:  # if sync transports
    uamqp_transport = kwargs.pop("uamqp_transport", False)
    if uamqp_transport and not UamqpTransport:
      raise ValueError("...import uamqp...")
    self._amqp_transport = UamqpTransport if uamqp_transport else PyamqpTransport
```
Although I might've been making it more complicated than necessary?

Reply:
This seems complicated...
I just can't see why we're tracking the boolean, and why we're checking sync vs async here?

Reply:
TL;DR: I removed this line, and in line 218 of `_client_base_async.py`, I changed `uamqp_transport = kwargs.pop("uamqp_transport", False)` to `uamqp_transport = kwargs.get("uamqp_transport", False)`. This way, the boolean is not passed in explicitly. It's passed in implicitly in `kwargs`. Answering your questions below.

1) We're tracking the boolean (i.e. passing it into `Configuration`) so that we can set two properties in the configuration object: 1) hostname and 2) custom_endpoint_address. I'm not sure if this is what you were thinking, but we're not setting a new `uamqp_transport` property in Configuration (i.e. we use `uamqp_transport` locally in the Configuration constructor, and don't set self.uamqp_transport).
2) In the snippet above, I assumed that `uamqp_transport` was not being passed in inside the ClientBaseAsync constructor to super(). So, just in that snippet, I was checking if the amqp_transport was passed in, b/c that would tell us that it was being called by ClientBaseAsync. And if that was the case, then we would still need to set `uamqp_transport` based on the type of the `amqp_transport` (to later be passed into Configuration). The only reason we were checking sync vs. async is that we'd need to set uamqp_transport if async. If sync, we can call `kwargs.pop("uamqp_transport", False).

Reply:
That sounds like it really needs a refactor! :D
But it's not public, so we don't need to block the PR on it.

Reply:
added to issue #26226